### PR TITLE
Add a per-PR view

### PR DESCRIPTION
**Pull request by driazati**
This is inspired by https://prow.k8s.io/, which provides an external page for information about PRs. We can add basically whatever we want here which frees us from being locked into GitHub comments (e.g. with Dr. CI). For now it's pretty basic and just shows artifacts (both on GitHub and in S3) for GitHub Action jobs for the pr at `hud.pytorch.org/pr/<number>`.

There's also a bit to get authenticated with GitHub via OAuth, which from what I can tell requires something like https://github.com/prose/gatekeeper. This is spun up in https://github.com/pytorch/test-infra/pull/24 (and already manually deployed)

This also pulls in bootstrap and makes the default font sans-serif site-wide

Fixes https://github.com/pytorch/pytorch/issues/59403

image

### CODE REVIEW

**Comment by ezyang** [comment]
RIP serverless hud 2021 ;)

File: src/PrDisplay.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+            );
+          }
+        }
+
+        // If there were any artifacts, set up the 'div' to show them
+        let artifactsElement = <div></div>;
+        if (artifacts.length > 0) {
+          artifactsElement = (
+            <div style={{ padding: "6px" }}>
+              <h5>Aritifacts</h5>
```
Comment (janeyx99) [inline src/PrDisplay.js:223]:
```suggestion
              <h5>Artifacts</h5>
```

File: src/App.js [hunk 2]
```diff
@@ -10,8 +10,10 @@ import QueueDisplay from "./QueueDisplay.js";
 import BuildHistoryDisplay from "./BuildHistoryDisplay.js";
 import GitHubStatusDisplay from "./GitHubStatusDisplay.js";
 import PerfHistoryDisplay from "./PerfHistoryDisplay.js";
+import PrDisplay from "./PrDisplay.js";
 import JobCorrelationHeatmap from "./JobCorrelationHeatmap.js";
 import GitHubActionsDisplay from "./GitHubActionsDisplay.js";
+import AuthorizeGitHub from "./GithubAuthorize.js";
```
Comment (janeyx99) [inline src/App.js:16]: tiny nit but maybe rename to GitHubAuthorize.js for consistency?
```suggestion
import AuthorizeGitHub from "./GitHubAuthorize.js";
```

File: src/PrDisplay.js [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    await asyncAll(promises);
+    this.setState(this.state);
+  }
+
+  render() {
+    let runs = undefined;
+    if (this.state.runs) {
+      runs = [];
+
+      // Render all of the check runs as a
```
Comment (janeyx99) [inline src/PrDisplay.js:148]: incomplete comment or do you actually mean a?

File: src/PrDisplay.js [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+                <span>({formatBytes(args.size_in_bytes)})</span>
+              </div>
+            );
+          }
+        }
+
+        // List out artifacts hosted on GitHub
+        if (run.artifacts) {
+          for (const [index, artifact] of run.artifacts.artifacts.entries()) {
+            // The URL in the is for the API, not browsers, so make it manually
```
Comment (janeyx99) [inline src/PrDisplay.js:185]:
```suggestion
            // The URL in the payload is for the API, not browsers, so make it manually
```

**Review by janeyx99 — COMMENTED** [review]
This is real cool! :D I'm not as familiar with JS to review the bigger functionality, but left some nits.

**Review by seemethere — APPROVED** [review]

**Comment by walterddr** [comment]
NICE.. I have a stretch ask --> it is possible to show the historic PR versions and their associated CI results?

**Comment by driazati** [comment]
> NICE.. I have a stretch ask --> it is possible to show the historic PR versions and their associated CI results?

I'm not sure if GitHub exposes workflow run IDs for previous commits (force pushes make this tricky) via their API, if so this is definitely do-able (just change the query to look back into the commit history instead of the most recent one). 

There's a ton of stuff we could add here (log views, Dr. CI stuff, perf testing charts), we should discuss a bit what info is most relevant and focus on that first.

**Pull request merged**